torch/serialization.py

```diff
@@ -256,6 +266,51 @@ class safe_globals(_weights_only_unpickler._safe_globals):
     """
 
 
+class skip_data:
+    """
+    Context-manager that skips writing storages for ``torch.save`` calls.
+
+    Space will still be reserved for storages by ``torch.save``, but storage bytes will not be written.
+
+    ..warning::
+        The ``skip_data`` context manager is an early prototype and is subject to change.
+
+    Args:
+        materialize_fake_tensors: Whether to materialize FakeTensors.
+
+    Example:
+        >>> import tempfile
+        >>> t = torch.randn(2, 3)
+        >>> with tempfile.NamedTemporaryFile() as f:
+        >>>     with torch.serialization.skip_data():
+        >>>         torch.save(t, f.name)
+        >>>     torch.load(f.name, weights_only=True)
+    """
+
+    def __init__(self, materialize_fake_tensors: bool = False):
+        self._old_skip_data = getattr(_serialization_tls, "skip_data", False)
+        self._old_materialize_fake_tensors = getattr(
+            _serialization_tls, "materialize_fake_tensors", False
+        )
+        self.materialize_fake_tensors = materialize_fake_tensors
```

Comment:
If this is context-manager-only, then everything should be in the enter really.
If it should be a global call as well then the code setting the values below need to be moved up here.